src/interface_elements.jl

```diff
@@ -0,0 +1,546 @@
+# and not an `InterfaceCellInterpolation{VectorizedInterpolation,VectorizedInterpolation}`.
+# To create a `VectorizedInterpolation{InterfaceCellInterpolation}`, `InterfaceCellInterpolation` needs to be a `ScalarInterpolation`.
+
+"""
+    InterfaceCellInterpolation(here::ScalarInterpolation, there::ScalarInterpolation) <: ScalarInterpolation
+
+An `InterfaceCellInterpolation` is an interpolation based on two interpolations on the faces of an `InterfaceCell`.
+If only one interpolation is given, it will be used for both faces.
+
+# Fields
+- `here::ScalarInterpolation`: interpolation on the face "here"
+- `there::ScalarInterpolation`: interpolation on the face "there"
+"""
+struct InterfaceCellInterpolation{sdim, rdim, shape<:AbstractRefShape{sdim}, rshape<:AbstractRefShape{rdim},
+        IPhere<:ScalarInterpolation{rshape}, IPthere<:ScalarInterpolation{rshape}} <: ScalarInterpolation{shape, Nothing}
+    here::IPhere
+    there::IPthere
+end
+
+function InterfaceCellInterpolation(here::IPhere, there::IPthere) where {
+        rdim, rshape<:AbstractRefShape{rdim}, IPhere<:ScalarInterpolation{rshape}, IPthere<:ScalarInterpolation{rshape}}
+    sdim, shape = get_interface_cell_sdim(rshape), get_interface_cell_shape(rshape)
+    return InterfaceCellInterpolation{sdim, rdim, shape, rshape, IPhere, IPthere}(here, there)
+end
+
+function InterfaceCellInterpolation(ip::ScalarInterpolation)
+    return InterfaceCellInterpolation(ip, ip)
+end
+
+getnbasefunctions(ip::InterfaceCellInterpolation) = getnbasefunctions(ip.here) + getnbasefunctions(ip.there)
+
+"""
+    get_n_dofs_on_side(get_dofs::Function, ip::InterfaceCellInterpolation, side::Symbol)
+
+Return the number of DOFs on a `side` (`:here` or `:there`) of an `InterfaceCellInterpolation`.
+The function `get_dofs` specifies which DOFs are considered, e.g. by passing `vertexdof_indices`.
+"""
+function get_n_dofs_on_side(get_dofs::Function, ip::InterfaceCellInterpolation, side::Symbol)
+    baseip = getproperty(ip, side)
+    return length(get_dofs(baseip)) > 0 ? sum(length.(get_dofs(baseip))) : 0
+end
+function get_n_dofs_on_side(get_dofs::Function, ip::VectorizedInterpolation{<:Any,<:Any,<:Any,<:InterfaceCellInterpolation}, side::Symbol)
+    baseip = getproperty(ip.ip, side)
+    return length(get_dofs(baseip)) > 0 ? sum(length.(get_dofs(baseip)))*n_components(ip) : 0
+end
+
+"""
+    get_interface_index(ip::InterfaceCellInterpolation, side::Symbol, i::Integer)
+
+Return the base function index for an `InterfaceCellInterpolation` given a `side` (`:here` or `:there`)
+and the local base function index `i` on that face.
+"""
+function get_interface_index(ip::InterfaceCellInterpolation, side::Symbol, i::Integer)
+    nvhere, nvthere = get_n_dofs_on_side(vertexdof_indices, ip, :here), get_n_dofs_on_side(vertexdof_indices, ip, :there)
+    nfhere, nfthere = get_n_dofs_on_side(facedof_interior_indices, ip, :here), get_n_dofs_on_side(facedof_interior_indices, ip, :there)
```

Comment:
I think we should consider defining empty defaults such as
```julia
Ferrite.facedof_interior_indices(::Lagrange{RefLine,1}) = ((), ())
```
instead of introducing branches. This would just mean that a line does not have interior facedofs. Imo that's easier to understand than branches.

(Introducing this fixes the problem for 2d dof distribution reported above).